pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -301,6 +309,7 @@ func NewController(client kubernetes.Interface, metadataClient metadata.Interfac
 
 	c.services = cache.NewSharedIndexInformer(svcMlw, &v1.Service{}, options.ResyncPeriod,
 		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+	c.serviceLister = listerv1.NewServiceLister(c.services.GetIndexer())
```

Comment:
@dgn @bison ^

Reply:
If this is just in reference to the lister, it looks fine. As long as the lister is based on an informer using the MLW, it should work like we expect it to.